### PR TITLE
Add code snippet for auto-detect language feature.

### PR DESCRIPTION
**Pull request by mnadev**
## Add code snippet for auto-detect language feature.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ X ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by nitsanshai** [comment]
maybe add a test? also can we name the files similarly to the others? maybe transcribe_chirp_auto_detect_language?

**Comment by mnadev** [comment]
> maybe add a test? also can we name the files similarly to the others? maybe transcribe_chirp_auto_detect_language?
> 
> in general following this format: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/speech/snippets/transcribe_chirp.py

Done

**Comment by nitsanshai** [comment]
Can we add a separate test file? Also looks like missing a header in the python file

File: speech/snippets/transcribe_chirp_auto_detect_language.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from google.api_core.client_options import ClientOptions
+from google.cloud.speech_v2 import SpeechClient
+from google.cloud.speech_v2.types import cloud_speech
+
+
+def transcribe_with_auto_detect_lang(
```
[the remark this answers is not in the document]
Comment (mnadev) [reply speech/snippets/transcribe_chirp_auto_detect_language.py]: Done.

**Comment by mnadev** [comment]
> Can we add a separate test file? Also looks like missing a header in the python file

Done

File: speech/snippets/transcribe_chirp_auto_detect_language.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+from google.cloud.speech_v2 import SpeechClient
+from google.cloud.speech_v2.types import cloud_speech
+
+
+def transcribe_chirp_auto_detect_language(
+    project_id: str,
+    audio_file: str,
+    region: str = "us-central1",
+) -> cloud_speech.RecognizeResponse:
+    """Transcribe an audio file and auto-detect spoken language using Chirp."""
```
Comment (m-strzelczyk) [inline speech/snippets/transcribe_chirp_auto_detect_language.py:30]: Would be nice to include info on requirements for the audio in the docstring. What format, what bitrate, max length etc.

**Review by m-strzelczyk — APPROVED** [review]

File: speech/snippets/transcribe_chirp_auto_detect_language.py (hunk 2 above)
Comment (mnadev) [reply speech/snippets/transcribe_chirp_auto_detect_language.py]: Done

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**